gluten-core/src/main/scala/io/glutenproject/extension/RemoveKnownFloatingPointNormalized.scala

```diff
@@ -0,0 +1,82 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.extension
+
+import io.glutenproject.extension.columnar.TransformHints
+
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.optimizer.NormalizeNaNAndZero
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.aggregate.{BaseAggregateExec, HashAggregateExec, ObjectHashAggregateExec, SortAggregateExec}
+
+/**
+ * [[KnownFloatingPointNormalized]] in Gluten is useless. When converting plans, it directly takes
+ * its child. We can reverse the KnownFloatingPointNormalized added by
+ * NormalizeFloatingNumbers.normalize in Gluten without affecting the results. Otherwise, pulling
+ * out pre-project in the logical plan is not sufficient. The KnownFloatingPointNormalized will be
+ * added to the groupingExpressions of the aggregate in the physical plan.
+ */
+object RemoveKnownFloatingPointNormalized extends Rule[SparkPlan] {
+
+  private def normalizeExpressionExists(expressions: Seq[Expression]): Boolean = {
+    expressions.exists(_.find(_.isInstanceOf[KnownFloatingPointNormalized]).isDefined)
+  }
+
+  /**
+   * If the groupingExpressions of the aggregate are expressions, an Alias will be added on the
+   * outside. After removing the KnownFloatingPointNormalized, it is necessary to remove this
+   * redundant Alias.
+   */
+  private def removeRedundantAlias(expression: Expression): Expression = expression match {
+    case _ @Alias(child: Attribute, _) => child
+    case other => other
+  }
+
+  private def supportTransform(plan: SparkPlan): Boolean =
+    TransformHints.isAlreadyTagged(plan) && TransformHints.isTransformable(plan)
+
+  override def apply(plan: SparkPlan): SparkPlan = applyWithValidation(plan)
+
+  def applyWithValidation(plan: SparkPlan, validation: Boolean = false): SparkPlan =
+    plan.transform {
+      // Only remove KnownFloatingPointNormalized when agg is transformable.
+      case agg: BaseAggregateExec
+          if normalizeExpressionExists(agg.groupingExpressions) &&
+            (validation || supportTransform(agg)) =>
+        val newGroupingExprs = agg.groupingExpressions.toIndexedSeq.map {
+          expr =>
+            val transformedExpr = expr.transform {
+              case KnownFloatingPointNormalized(NormalizeNaNAndZero(child)) => child
+              case KnownFloatingPointNormalized(If(IsNull(expr), _, _)) => expr
+              case KnownFloatingPointNormalized(ArrayTransform(expr, _)) => expr
+            }
+            removeRedundantAlias(transformedExpr).asInstanceOf[NamedExpression]
+        }
+        agg match {
+          case hash: HashAggregateExec =>
+            hash.copy(groupingExpressions = newGroupingExprs)
```

Comment:
It seems we should fail in `TransformPreOverrides` as  we apply `RemoveKnownFloatingPointNormalized` first and do not copy tags, but the tests passed.. can you check if this rule is actually applied ?
https://github.com/oap-project/gluten/blob/451b1cee89b9425443924ae69f9dd04a186c85a4/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L282C1-L283